### PR TITLE
node 6.0.0

### PR DESCRIPTION
**Pull request by xeoneux**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
### Description

Update node to v6.0.0

### CODE REVIEW

File: Formula/node.rb [hunk 1]
```diff
@@ -11,11 +11,11 @@ class Node < Formula
     sha256 "8254c09259aa2d869279f580e489d82181aec17b7555c2f434c290e2625906d1" => :mavericks
   end
 
-  devel do
-    url "https://nodejs.org/download/rc/v6.0.0-rc.3/node-v6.0.0-rc.3.tar.xz"
-    sha256 "472ae46a205fb65cd784b22825b7354cddb4da5f46c1974272be43e55745ce6e"
-    version "6.0.0-rc.3"
-  end
+  # devel do
```
Comment (dunn) [inline Formula/node.rb]: Just remove this block entirely if there's no new development release.

File: Formula/node.rb [hunk 2]
```diff
@@ -47,7 +41,7 @@ class Node < Formula
   end
 
   resource "icu4c" do
-    url "https://ssl.icu-project.org/files/icu4c/56.1/icu4c-56_1-src.tgz"
+    url "https://ssl.icu-project.org/files/icu4c/57.1/icu4c-57_1-src.tgz"
     mirror "https://ftp.mirrorservice.org/sites/download.qt-project.org/development_releases/prebuilt/icu/src/icu4c-56_1-src.tgz"
```
Comment (dunn) [inline Formula/node.rb]: The mirror needs to be updated too.

File: Formula/node.rb [hunk 3]
```diff
@@ -47,7 +41,7 @@ class Node < Formula
...
     version "56.1"
```
Comment (dunn) [inline Formula/node.rb]: Also the version number and `sha256`.

**Comment by xeoneux** [comment]
@dunn `MACOSX_DEPLOYMENT_TARGET` changed to 10.7. Does it affect the version requirement?

**Comment by xeoneux** [comment]
All tests pass except `azure-cli` due to changes in the Node.js v6.0.0 API. [Issue](https://github.com/Azure/azure-xplat-cli/issues/2820) at [Azure/azure-xplat-cli](https://github.com/Azure/azure-xplat-cli).

**Comment by dunn** [comment]
Thanks for opening the Azure PR!  Hopefully they'll merge it quickly.

**Comment by chrmoritz** [comment]
The ICU 57 upgrade didn't made it into node v6.0.0, so it should be reverted from this PR (as it's not supported).

Also there are no compatibility issue with azure-cli and node v6. It just need the be recompiled against the new v8 5.0 shiped with node 6. You have to simply add a revision to the azure-cli formula to do this.

**Comment by jbergstroem** [comment]
How about introducing support for building against a shared c-ares?

**Comment by chrmoritz** [comment]
@jbergstroem: Is this comment https://github.com/nodejs/node/pull/5775#issuecomment-199143852 still valid? If not adding an option to build against a shared c-ares shouldn`t be a problem.

**Comment by jbergstroem** [comment]
@chrmoritz yes, I've landed shared support against c-ares. Additionally, if you want, building against a shared zlib and http-parser should work fine as well.

**Comment by xeoneux** [comment]
Azure/azure-xplat-cli#2821 got merged into base branch. Are we waiting for a release?

**Comment by xeoneux** [comment]
Also, check Homebrew/homebrew-core#638 for my motivation to keep npm packages out of homebrew.

**Comment by DomT4** [comment]
> Azure/azure-xplat-cli#2821 got merged into base branch. Are we waiting for a release?

No need to wait. Can backport the patch with a link to the upstream commit and explanation it can be removed on next stable release, or simply use the commit diff as a `patch do` if it applies cleanly with the older release.

**Comment by LinusU** [comment]
@DomT4 Can we merge this now then?

**Comment by DomT4** [comment]
Seems reasonable. I'll deal with the `azure-cli` issue shortly after merging.

Merged in https://github.com/Homebrew/homebrew-core/commit/b8b57c7652338d0bebf0a08a42ccb6985e914f11. Thank you for your contribution to Homebrew @xeoneux; we appreciate it! :smiley_cat:

**Comment by LinusU** [comment]
Awesome 🎉

**Pull request closed without merging**